app/assets/javascripts/components/NotesList.tsx

```diff
@@ -0,0 +1,118 @@
+import { KeyboardKey } from '@/services/ioService';
+import { AppState } from '@/ui_models/app_state';
+import { DisplayOptions } from '@/ui_models/app_state/notes_view_state';
+import { SNNote } from '@standardnotes/snjs';
+import { observer } from 'mobx-react-lite';
+import { FunctionComponent } from 'preact';
+import { NotesListItem } from './NotesListItem';
+import { toDirective } from './utils';
+
+type Props = {
+  appState: AppState;
+  notes: SNNote[];
+  selectedNotes: Record<string, SNNote>;
+  displayOptions: DisplayOptions;
+  paginate: () => void;
+};
+
+const FOCUSABLE_BUT_NOT_TABBABLE = -1;
+const NOTES_LIST_SCROLL_THRESHOLD = 200;
+
+export const NotesList: FunctionComponent<Props> = observer(
+  ({ appState, notes, selectedNotes, displayOptions, paginate }) => {
+    const { selectPreviousNote, selectNextNote } = appState.notesView;
+    const { hideTags, hideDate, hideNotePreview, sortBy } = displayOptions;
+
+    const tagsStringForNote = (note: SNNote): string => {
+      if (hideTags) {
+        return '';
+      }
+      const selectedTag = appState.selectedTag;
+      if (!selectedTag) {
+        return '';
+      }
+      if (selectedTag.isSmartTag) {
+        return appState
+          .getNoteTags(note)
+          .map((tag) => '#' + tag.title)
+          .join(' ');
+      }
+      const tags = appState.getNoteTags(note);
+      if (tags.length === 1) {
+        return '';
+      }
+      return tags.map((tag) => '#' + tag.title).join(' ');
+    };
```

Comment:
Something looks fishy with this code, it's probably on me but it looks like we're saying:

- if we have a smart tag: we dump the list of tags
- if we have a regular tag: we ignore the list of tags is there is only one, else we dump the list of tags.

I'd assume both case should do the same, maybe:

- if we have a smart tag: we ignore the list of tags is there is only one, else we dump the list of tags.
- if we have a regular tag: we ignore the list of tags is there is only one, else we dump the list of tags.

Could we do something like:

```ts
const tagTitleWithHash = tag => `#${tag.title}`
appState
   .getNoteTags(note)
   .filter(tag => tag.uuid !== selectedTag.uuid)
   .map(tagTitleWithHash)
   .join(' ')
```

?